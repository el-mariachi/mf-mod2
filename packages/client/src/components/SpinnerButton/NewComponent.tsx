export default function NewComponent() {
  return (
    <>
      <h1>New component</h1>
    </>
  )
}
