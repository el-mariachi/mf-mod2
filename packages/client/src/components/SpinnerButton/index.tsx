import { FC, ButtonHTMLAttributes } from 'react'
import { Button } from 'react-bootstrap'
import { Spinner } from 'react-bootstrap'

export type SpinnerButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  loading: boolean
  spinnerText?: string
}
const SpinnerButton: FC<SpinnerButtonProps> = ({
  children,
  spinnerText,
  loading,
  ...props
}) => {
  if (!spinnerText) {
    spinnerText = 'Секундочку'
  }
  return (
    <Button type="submit" disabled={loading} {...props}>
      {loading ? (
        <>
          <Spinner
            as="span"
            animation="border"
            size="sm"
            role="status"
            aria-hidden="true"
          />
          {` ${spinnerText}...`}
        </>
      ) : (
        children
      )}
    </Button>
  )
}
export default SpinnerButton
