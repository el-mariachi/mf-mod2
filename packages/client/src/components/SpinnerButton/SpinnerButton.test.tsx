import SpinnerButton from './index'
import { render, screen } from '@testing-library/react'

describe('#SpinnerButton', () => {
  it('SpinnerButton should to match snapshot', () => {
    const { baseElement } = render(<SpinnerButton loading />)
    expect(baseElement).toMatchSnapshot()
  })
  it('should create', () => {
    const renderedSpinnerButton = render(<SpinnerButton loading />)
    expect(renderedSpinnerButton).toBeTruthy()
  })
  it('should be rendered with default text if text not provided in props', () => {
    render(<SpinnerButton loading />)
    expect(screen.getByRole('button')).toHaveTextContent('Секундочку')
  })
  it('should be rendered with text from props', () => {
    render(<SpinnerButton loading spinnerText="Минуточку" />)
    expect(screen.getByRole('button')).toHaveTextContent('Минуточку')
  })
  it('should be disabled if loading true', () => {
    render(<SpinnerButton loading />)
    expect(screen.getByRole('button')).toHaveAttribute('disabled')
  })
})
