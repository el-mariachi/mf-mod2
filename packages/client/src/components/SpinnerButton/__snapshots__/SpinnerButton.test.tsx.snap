// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#SpinnerButton SpinnerButton should to match snapshot 1`] = `
<button
  className="btn btn-primary"
  disabled={true}
  type="submit"
>
  <span
    aria-hidden="true"
    className="spinner-border spinner-border-sm"
    role="status"
  />
   Секундочку...
</button>
`;
